src/dotnet/CommandLine/CommandLineApplication.cs

```diff
@@ -334,7 +340,15 @@ public void ShowHelp(string commandName = null)
             for (var cmd = this; cmd != null; cmd = cmd.Parent)
             {
                 cmd.IsShowingInformation = true;
-                headerBuilder.Insert(6, string.Format(" {0}", cmd.Name));
+                if (cmd != this && cmd.Arguments.Any())
+                {
+                    var args = string.Join(" ", cmd.Arguments.Select(arg => arg.Name));
+                    headerBuilder.Insert(6, string.Format(" {0} {1}", cmd.Name, args));
```

Comment:
It comes from this: var headerBuilder = new StringBuilder("Usage:"); I can change the magic number 6 to be headerBuilder.Length